src/services/transaction_reversal.rs

```diff
@@ -15,17 +20,199 @@ pub struct TransactionReversalPayload<'mpesa> {
-    queue_timeout_url: &'mpesa str,
+    timeout_url: Option<&'mpesa str>,
     #[serde(rename(serialize = "Remarks"))]
-    remarks: &'mpesa str,
+    remarks: Option<&'mpesa str>,
     #[serde(rename(serialize = "Occasion"))]
-    ocassion: &'mpesa str,
+    occasion: Option<&'mpesa str>,
+    #[serde(rename(serialize = "Amount"))]
+    amount: f64,
 }
 
-pub struct TransactionReversalBuilder;
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TransactionReversalResponse {
+    #[serde(rename(deserialize = "ConversationID"))]
+    conversation_id: String,
+    #[serde(rename(deserialize = "OriginatorConversationID"))]
+    originator_conversation_id: String,
+    #[serde(rename(deserialize = "ResponseDescription"))]
+    response_description: String,
+}
+
+#[derive(Debug)]
+pub struct TransactionReversalBuilder<'mpesa, Env: ApiEnvironment> {
+    client: &'mpesa Mpesa<Env>,
+    initiator: &'mpesa str,
+    command_id: Option<CommandId>,
+    transaction_id: Option<&'mpesa str>,
+    receiver_party: Option<&'mpesa str>,
+    receiver_identifier_type: Option<&'mpesa str>,
+    result_url: Option<&'mpesa str>,
+    timeout_url: Option<&'mpesa str>,
+    remarks: Option<&'mpesa str>,
+    occasion: Option<&'mpesa str>,
+    amount: Option<f64>,
+}
+
+impl<'mpesa, Env: ApiEnvironment> TransactionReversalBuilder<'mpesa, Env> {
+    /// Creates new `TransactionReversalBuilder`
+    pub fn new(
+        client: &'mpesa Mpesa<Env>,
+        initiator: &'mpesa str,
+    ) -> TransactionReversalBuilder<'mpesa, Env> {
+        TransactionReversalBuilder {
+            client,
+            initiator,
+            command_id: None,
+            transaction_id: None,
+            receiver_party: None,
+            receiver_identifier_type: None,
+            result_url: None,
+            timeout_url: None,
+            remarks: None,
+            occasion: None,
+            amount: None,
+        }
+    }
+
+    /// Adds `CommandId`. Defaults to `CommandId::TransactionReversal` if no value explicitly passed
+    ///
+    /// # Errors
+    /// If `CommandId` is not valid
+    pub fn command_id(mut self, command_id: CommandId) -> Self {
+        self.command_id = Some(command_id);
+        self
+    }
+
+    /// Add the Mpesa Transaction ID of the transaction which you wish to reverse
+    ///
+    /// This is a required field.
+    pub fn transaction_id(mut self, transaction_id: &'mpesa str) -> Self {
+        self.transaction_id = Some(transaction_id);
+        self
+    }
+
+    /// Organization receiving the transaction
+    ///
+    /// This is required field
+    pub fn receiver_party(mut self, receiver_party: &'mpesa str) -> Self {
+        self.receiver_party = Some(receiver_party);
+        self
+    }
+
+    /// Type of organization receiving the transaction
+    ///
+    /// This is required field
+    pub fn receiver_identifier_type(mut self, receiver_identifier_type: &'mpesa str) -> Self {
```

Comment:
Reciever identifier types are represented as an enum `IdentifierType` with the default for this api being `IdentifierType::ShortCode`

```suggestion
    pub fn receiver_identifier_type(mut self, receiver_identifier_type: IdentifierType) -> Self {
```

Reply:
Let me work on the fix